### ISSUE TITLE
[bug] Syntax errors when reading JSON

### ISSUE THREAD

**Issue by ethancarlsson** [pfeiferj/tree-sitter-hurl#5]
Hi,

I recently installed this via nvim-treesitter (thanks for your work on this by the way), and it seems to be identifying syntax errors incorrectly when using JSON. 

For example (red is a syntax error)
<img width="356" alt="image" src="https://github.com/pfeiferj/tree-sitter-hurl/assets/63494553/792b5cfb-b01a-4329-9874-64c7c64023cb">

I think one of the issues might be a conflict with how the tree-sitter-json is injected.
```
% tree-sitter highlight examples/front_page/json_body.hurl 
Error in query file "injections.scm"

Caused by:
    Query error at 1:3. Invalid node type input
```

I wrote a quick test to see what was happening
```
==================
POST with JSON input
==================
POST https://example.org/api/dogs
{
    "id": 0,
    "name": "Frieda",
    "picture": "images/scottish-terrier.jpeg",
    "age": 3,
    "breed": "Scottish Terrier",
    "location": "Lisco, Alabama"
}

---

(hurl_file
	(entry
	)
)
```
and I noticed that this treesitter implementation seems to try and reimplement JSON.

```
...
          (body
            (bytes
              (json_value
                (json_object
                  (json_key_value
                    (json_key_string
                      (json_string_content
                        (json_string_text))
                      (json_string_content
                        (json_string_text)))
...
```
Wouldn't it be easier and more consistent to just detect that the input is json and then let the injected treesitter handle json, and other treesitters.

I also noticed that when I use multiline strings there are a some unexpected results that prevent the injection from working as expected..
```
==================
POST with graphql input
==================

POST https://swapi-graphql.netlify.app/.netlify/functions/index
\`\`\`graphql
query Query {
  allFilms {
    films {
      title
      director
      releaseDate
      speciesConnection {
        species {
          name
          classification
          homeworld {
            name
          }
        }
      }
    }
  }
}
\`\`\`
```
^ Reads as a multiline string instead of graphql

```
==================
POST with messy muliline string input
==================

POST https://example.org/api/dogs
\`\`\`
{"test": "this is the JSON"}
^ This is a multiline string but it has json in it.
\`\`\`
```
^ Interprets this as badly formed JSON, where I would expect it to be interpreted as a string.

I'd be willing to look into it, and make some pull requests if you'd like. If I do would you be willing to ditch the JSON parts and just rely on injections?

**Comment by pfeiferj** [comment]
@ethancarlsson Hey, sorry for the delay in getting back to you on this, not sure how i missed your comment. So the issue is that you have to be able to give a range for the injection to take over. So we have to detect when there is a json body. The hurl syntax is a little bit tricky for treesitter because a lot of the sections can be different things based on what comes before it and how it's formatted. For example, you can do a json body by simply opening and closing a json object or array right below the POST line, or you can do an xml body by simply creating an opening and closing xml element. so to detect the beginning and end of a json body we have to have at least enough ability to parse json to know if the starting brace gets closed and then allow the json injection to take over on that area.

All that being said I'm going to guess the issue here with the multiline string actually has something to do with the variable injection part of the treesitter query. My guess is that it's seeing the opening brace within the multiline string, trying to parse a variable injection, failing, thereby causing the multiline string parse to fail. Then treesitter tries to recover from the error and ends up deciding the ``` line was actually the bad line and the next line is in fact a json body. I'll try to take a look over that part of the query here tomorrow and see if my theory is correct/if I can come up with a solution

For the first example I'm pretty sure it has to do with the comment line handling. Comment lines have proven to be fairly difficult for similar reasons, with all the dynamic starts and ends to sections it's hard to have the query decide if something should be a comment. Treesitter queries have some extra logic for comments but unfortunately it causes issues with many of the section starts/ends. So you end up having to make the comment less greedy and then scattering it in with other parts of the query. But that leads to more places to get the behavior wrong. Unfortunately that means if it _is_ a comment line handling issue i'm much less confident i'll be able to find a fix than with the multiline string one. But I have a couple vague suspicions so i'll take a look after seeing what I can do about the multiline scenario

**Comment by pfeiferj** [comment]
These issues should be resolved in v1.4.1. If there's any edge cases I missed lets just open a new issue
